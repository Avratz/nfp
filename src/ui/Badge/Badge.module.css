.badge {
  border: 2px solid;
  text-transform: uppercase;
  font-weight: var(--font-bold);
  font-size: var(--text-xs);
  line-height: var(--line-xs);
  border-radius: 0.25rem;
  padding: 0.375rem;
  padding-top: 0.28rem;
  padding-bottom: 0.13rem;
}

.outline {
  color: var(--green);
  background-color: transparent;
  border-color: var(--green);
}

.solid {
  color: var(--neutral-8);
  background-color: var(--green);
  border-color: var(--green);
}
