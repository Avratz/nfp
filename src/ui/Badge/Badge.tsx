import { ReactNode } from 'react'
import { cva, VariantProps } from 'class-variance-authority'

import styles from './Badge.module.css'

const badgeStyles = cva(styles.badge, {
  variants: {
    variant: {
      outline: styles.outline,
      solid: styles.solid,
    },
  },
  defaultVariants: {
    variant: 'solid',
  },
})

export interface BadgeProps extends VariantProps<typeof badgeStyles> {
  children: ReactNode
  className?: string
}

export function Badge({ variant, children, className }: BadgeProps) {
  return (
    <div
      className={badgeStyles({
        variant,
        className,
      })}
    >
      {children}
    </div>
  )
}
