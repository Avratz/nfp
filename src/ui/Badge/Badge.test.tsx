import { render } from '@testing-library/react'
import { Badge } from './Badge'

describe('Avatar', () => {
  it('renders outline', () => {
    const { container } = render(<Badge variant="outline">outline</Badge>)
    expect(container).toMatchSnapshot()
  })

  it('renders solid', () => {
    const { container } = render(<Badge variant="solid">solid</Badge>)
    expect(container).toMatchSnapshot()
  })
})
