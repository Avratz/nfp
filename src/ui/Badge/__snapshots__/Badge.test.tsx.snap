// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar renders outline 1`] = `
<div>
  <div
    class="badge outline"
  >
    outline
  </div>
</div>
`;

exports[`Avatar renders solid 1`] = `
<div>
  <div
    class="badge solid"
  >
    solid
  </div>
</div>
`;
